app/storage/metadata_parser.py

```diff
@@ -1,103 +1,106 @@
-
-def boolean_parser(boolean_value):
-    if not isinstance(boolean_value, bool):
-        raise TypeError('Claim was not of type `bool`')
-    return boolean_value
-
-
-def string_parser(string_value):
-    if not isinstance(string_value, str) or string_value == '':
-        raise TypeError('Claim not a valid string value')
-    return string_value
-
+class UUIDString(fields.UUID):
+    """ Currently, runner cannot handle UUID objects in metadata
+    Since all metadata is serialised and deserialised to JSON.
+    This custom field deserialises UUIDs to strings.
+    """
 
-def optional_string_parser(optional_string_value):
-    if not isinstance(optional_string_value, str):
-        raise TypeError('Claim not a valid/empty string')
-    return optional_string_value
+    def _deserialize(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        return str(super()._deserialize(*args, **kwargs))
 
 
-def clean_leading_trailing_spaces(metadata):
-    for key, value in metadata.items():
-        if isinstance(value, str):
-            metadata[key] = value.strip()
+class DateString(fields.DateTime):
+    """ Currently, runner cannot handle Date objects in metadata
+    Since all metadata is serialised and deserialised to JSON.
+    This custom field deserialises Dates to strings.
+    """
 
-    return metadata
+    def _deserialize(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        return super()._deserialize(*args, **kwargs).strftime('%Y-%m-%d')
 
 
 VALIDATORS = {
-    'date': iso_8601_date_parser,
-    'uuid': uuid_4_parser,
-    'boolean': boolean_parser,
-    'string': string_parser,
-    'optional_string': optional_string_parser,
+    'date': functools.partial(DateString, format='%Y-%m-%d', required=True),
+    'uuid': functools.partial(UUIDString, required=True),
+    'boolean': functools.partial(fields.Boolean, required=True),
+    'string': functools.partial(fields.String, required=True),
+    'url': functools.partial(fields.Url, required=True),
 }
 
-MANDATORY_METADATA = [
-    {'name': 'eq_id', 'validator': 'string'},
-    {'name': 'form_type', 'validator': 'string'},
-    {'name': 'ru_ref', 'validator': 'string'},
-    {'name': 'collection_exercise_sid', 'validator': 'string'},
-    {'name': 'tx_id', 'validator': 'uuid'},
-    {'name': 'case_id', 'validator': 'uuid'},
-    {'name': 'response_id', 'validator': 'string'},
-]
 
+class StripWhitespaceMixin:
+    @pre_load
+    def strip_whitespace(self, items):  # pylint: disable=no-self-use
+        for key, value in items.items():
+            if isinstance(value, str):
+                items[key] = value.strip()
+        return items
 
-def parse_runner_claims(claims):
-    cleaned_claims = clean_leading_trailing_spaces(claims.copy())
-    validate_metadata(cleaned_claims, MANDATORY_METADATA)
 
-    return cleaned_claims
+class RunnerMetadataSchema(Schema, StripWhitespaceMixin):
+    """ Metadata which is required for the operation of runner itself
+    """
```

Comment:
**Critical**, PEP recommends single line docstrings to be closing brackets on the same line. 😑  👀